context.AfterFunc is not included in Go version 1.20.x , but used in this library.

Thank you so much for developing this library. While I'm trying this library, There's a problem with test, so please let me an report.  I'm sorry if i was wrong.

**Describe the bug**
This library's go version is 1.20 (or later) according to go.mod, But context.AfterFunc is used in Connect function in paho/client.go .

But, [context.AfterFunc is not included inGo version 1.20.x](https://pkg.go.dev/context#AfterFunc). It has been offered from version 1.21.x .

**To reproduce**
After you switch to Go version 1.20.x and make test, you can get an error like the following.
```
$ go version
go version go1.20.13 linux/amd64

$ make test
go test -coverprofile /tmp/autopaho_coverage.out -race -tags=unittest ./autopaho/ -v -count 1
# github.com/eclipse/paho.golang/paho
paho/client.go:337:10: undefined: context.AfterFunc
FAIL    github.com/eclipse/paho.golang/autopaho [build failed]
FAIL
make: *** [Makefile:4: unittest] Error 1
```

**Expected behaviour**
Test should be passed

**Additional context**
The specified Go version in this library probably is required upgrading.

Best Regards,

--- a/go.mod
+++ b/go.mod
@@ -1,14 +1,13 @@
 module github.com/eclipse/paho.golang
 
-go 1.20
+go 1.21
 
 require (
 	github.com/google/go-cmp v0.5.9
-	github.com/gorilla/websocket v1.5.0
+	github.com/gorilla/websocket v1.5.1
 	github.com/stretchr/testify v1.8.4
 	go.uber.org/goleak v1.2.1
-	golang.org/x/net v0.17.0
-	golang.org/x/sync v0.4.0
+	golang.org/x/net v0.21.0
 )
 
 require (

